### PR TITLE
expose kubernetes service cluster ip/external name

### LINKED ISSUE #3110: Get the clusterIP from service via kubernetes_sd meta label
**Environment**

Kubernetes v1.5.4+coreos.0"

* Prometheus version:

quay.io/prometheus/prometheus:v2.0.0-beta.0

* Prometheus configuration file:
```
    - job_name: 'kube-state-metrics'
      kubernetes_sd_configs:
      - role: service

      relabel_configs:
      - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_service_label_k8s_app]
        action: keep
        regex: prometheus;kube-state-metrics
```

**Question**

When I use `role: service` it use by default the  `svc.Name+"."+svc.Namespace+".svc"` (https://github.com/prometheus/prometheus/blob/master/discovery/kubernetes/service.go#L153) as address. Since we are not using skydns in our k8s cluster we would like to use the service ip / ClusterIP directly.

As the same way you get the port from spec (https://github.com/prometheus/prometheus/blob/master/discovery/kubernetes/service.go#L152), is it possible to get the clusterIP (when it is present) and export it as metalabel `__meta_kubernetes_service_clusterIP` to prometheus ?

### PR DESCRIPTION
**Pull request by tariq1890**
Creating this PR as https://github.com/prometheus/prometheus/pull/3275 has no activity. Closes #3110.

@simonpasquier @brian-brazil  FYI

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -700,6 +700,8 @@ Available meta labels:
 * `__meta_kubernetes_service_port_name`: Name of the service port for the target.
 * `__meta_kubernetes_service_port_number`: Number of the service port for the target.
 * `__meta_kubernetes_service_port_protocol`: Protocol of the service port for the target.
+* `__meta_kubernetes_service_cluster_ip`: The cluster IP address of the service. (Does not apply to services of type ExternalName)
+* `__meta_kubernetes_service_external_name`: The dns-resolvable of the service. (Applies to services of type ExternalName)
```
Comment (simonpasquier) [inline docs/configuration/configuration.md]: `The DNS name of the service.` instead of `The dns-resolvable of the service.`

**Comment by simonpasquier** [comment]
@tariq1890 FYI I've edited your description to reference the original issue.

File: docs/configuration/configuration.md (hunk 1 above)
Comment (tariq1890) [reply docs/configuration/configuration.md]: Done :)

**Comment by tariq1890** [comment]
@simonpasquier Sure!

**Comment by tariq1890** [comment]
@simonpasquier Done! Ready for another round of review :)

**Comment by tariq1890** [comment]
Can this be merged?

**Comment by simonpasquier** [comment]
We've just released the first release candidate of v2.6.0 so the master branch is currently frozen. It will have to wait for v2.6.0 to be released before merging.

**Comment by tariq1890** [comment]
Sure thanks for letting me know :)

**Comment by tariq1890** [comment]
Sure thanks for letting me know :)

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -700,6 +700,8 @@ Available meta labels:
 * `__meta_kubernetes_service_port_name`: Name of the service port for the target.
 * `__meta_kubernetes_service_port_number`: Number of the service port for the target.
 * `__meta_kubernetes_service_port_protocol`: Protocol of the service port for the target.
+* `__meta_kubernetes_service_cluster_ip`: The cluster IP address of the service. (Does not apply to services of type ExternalName)
```
Comment (brian-brazil) [inline docs/configuration/configuration.md]: keep it sorted

File: discovery/kubernetes/service.go [hunk 3]
```diff
@@ -135,11 +135,13 @@ func serviceSourceFromNamespaceAndName(namespace, name string) string {
...
-	serviceAnnotationPrefix  = metaLabelPrefix + "service_annotation_"
-	servicePortNameLabel     = metaLabelPrefix + "service_port_name"
-	servicePortProtocolLabel = metaLabelPrefix + "service_port_protocol"
+	serviceNameLabel          = metaLabelPrefix + "service_name"
+	serviceLabelPrefix        = metaLabelPrefix + "service_label_"
+	serviceAnnotationPrefix   = metaLabelPrefix + "service_annotation_"
+	servicePortNameLabel      = metaLabelPrefix + "service_port_name"
+	servicePortProtocolLabel  = metaLabelPrefix + "service_port_protocol"
+	serviceClusterIPLabel     = metaLabelPrefix + "service_cluster_ip"
+	serviceExternalNamePLabel = metaLabelPrefix + "service_external_name"
```
Comment (brian-brazil) [inline discovery/kubernetes/service.go]: why the P?

**Comment by tariq1890** [comment]
@brian-brazil Review comments addressed.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**